test/distributed/fsdp/test_fsdp_core.py

```diff
@@ -213,6 +206,79 @@ def test_param_change_after_init(self):
         )
 
 
+class TestSerialization(FSDPTest):
+    def _get_init_modes_for_test(self, cpu_offload):
+        # cpu offload does not support FSDPInitMode.CUDA_AFTER
+        if cpu_offload.offload_params:
+            return [FSDPInitMode.CUDA_NEVER, FSDPInitMode.CUDA_BEFORE]
+        # model should be placed on CUDA when cpu_offload is not enabled
+        else:
+            return [FSDPInitMode.CUDA_AFTER, FSDPInitMode.CUDA_BEFORE]
+
+    @skip_if_lt_x_gpu(2)
+    @parametrize(
+        "cpu_offload",
+        [CPUOffload(offload_params=True), CPUOffload(offload_params=False)],
+    )
+    def test_pickle(self, cpu_offload):
+        """
+        Ensure that wrapped modules can be pickled/unpickled.
+        """
+        init_modes = self._get_init_modes_for_test(cpu_offload)
+        for fsdp_init_mode in init_modes:
+            with self.subTest(fsdp_init_mode=fsdp_init_mode):
+                group = dist.distributed_c10d._get_default_group()
+                model = self._get_model(group, fsdp_init_mode, cpu_offload)
+                model = pickle.loads(pickle.dumps(model))
```

Comment:
Note that this is not an officially supported thing for plain nn.Module. You should be using state dict to serialize the models instead of serializing the whole object.
I am not even sure that this is tested for regular Modules so that might break in the future.